### PR TITLE
OEL-492: Rich text paragraph theming

### PR DESCRIPTION
**Pull request by abel-santos-corral**
Jira issue(s):
- [OEL-492](https://citnet.tech.ec.europa.eu/CITnet/jira/browse/OEL-492)

(!) Don't merge until OEL-824 (meaning BCL pattern v.0.15.0) is already at 1.x branch.

### CODE REVIEW

File: kits/default/package.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
     "production": "npm-run-all build:*"
   },
   "devDependencies": {
-    "@openeuropa/bcl-builder": "0.14.0",
+    "@openeuropa/bcl-builder": "0.15.0",
```
Comment (escuriola) [inline kits/default/package.json]: Do not upgrade to this version, is still on development and blocks this if we have to wait until v15 is out.

File: package.json [hunk 2]
```diff
@@ -13,8 +13,8 @@
     "production": "npm-run-all build:*"
   },
   "devDependencies": {
-    "@openeuropa/bcl-builder": "0.14.0",
-    "@openeuropa/bcl-theme-default": "0.14.0",
+    "@openeuropa/bcl-builder": "0.15.0",
```
Comment (escuriola) [inline package.json]: Same as above

File: tests/src/Kernel/Paragraphs/RichTextTest.php [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+ * Tests the "Rich text" paragraphs.
+ */
+class RichTextTest extends ParagraphsTestBase {
+
+  /**
+   * Tests the rendering of the paragraph type.
+   */
+  public function testRendering(): void {
+    // Create Rich text paragraph.
+    $text_original = 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque ornare et elit a dictum. Maecenas lacinia eros quis eros iaculis, sit amet bibendum massa facilisis. Integer arcu nisl, fringilla nec quam vel, tincidunt maximus ex. Suspendisse ac arcu efficitur, feugiat tellus vel, viverra sapien. Etiam vitae condimentum lorem. Nulla congue ligula lacinia efficitur tempus. Duis vitae auctor enim. Nulla iaculis, diam et sagittis scelerisque, est mauris luctus sem, a imperdiet lacus diam eu dui. Morbi accumsan, augue eu gravida elementum, libero mi blandit odio, eu fringilla nunc ipsum non tellus. Suspendisse dapibus elit at lobortis pretium. Quisque vestibulum ut purus sit amet molestie. Sed eget volutpat justo, vel varius augue. Vestibulum vel risus facilisis, feugiat sem aliquam, lobortis ante.';
```
Comment (escuriola) [inline tests/src/Kernel/Paragraphs/RichTextTest.php]: It's not possible to test more "html" is this text? I understand that rich text is text with some kind of format, as bold, italics, headings, underlined... and this is plain text.

**Review by escuriola — CHANGES_REQUESTED** [review]

**Review by escuriola — DISMISSED** [review]

File: templates/overrides/field/field--paragraph--field-oe-title--oe-rich-text.html.twig [hunk 4]
```diff
@@ -0,0 +1,10 @@
+{#
```
Comment (drishu) [inline templates/overrides/field/field--paragraph--field-oe-title--oe-rich-text.html.twig]: I see it was done the same in oe_theme but I'd rather have all this paragraphs theming in one place, and crate paragraph--oe-rich-text.html.twig with
```
<h4 class="fw-bold mb-4">{{ content.field_oe_title }}</h2>
{{ content|without('field_oe_title') }}
```
I added the same class as the one in listing item block, I see in netlify in paragraphs its the same used, so lets go with this for now

File: tests/src/Kernel/Paragraphs/ParagraphsTestBase.php [hunk 5]
```diff
@@ -17,7 +17,9 @@ abstract class ParagraphsTestBase extends AbstractKernelTestBase {
    */
   protected static $modules = [
     'allowed_formats',
+    'ckeditor',
```
Comment (drishu) [inline tests/src/Kernel/Paragraphs/ParagraphsTestBase.php]: you don't need these changes

**Review by drishu — CHANGES_REQUESTED** [review]

File: tests/src/Kernel/Paragraphs/RichTextTest.php [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+ * Tests the "Rich text" paragraphs.
+ */
+class RichTextTest extends ParagraphsTestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp(): void {
+    parent::setUp();
+    // Create a text format and associate CKEditor.
```
Comment (drishu) [inline tests/src/Kernel/Paragraphs/RichTextTest.php]: CKEditor is nowhere in this test, but even so, it is obvious what that code does so the comment is redundant

File: tests/src/Kernel/Paragraphs/RichTextTest.php [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+      'weight' => 0,
+    ]);
+    $this->filterFormat->save();
+  }
+
+  /**
+   * Tests the rendering of the paragraph type.
+   */
+  public function testRendering(): void {
+    // Create Rich text paragraph.
```
Comment (drishu) [inline tests/src/Kernel/Paragraphs/RichTextTest.php]: remove this comment

File: tests/src/Kernel/Paragraphs/RichTextTest.php [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+      'type' => 'oe_rich_text',
+      'field_oe_title' => 'Rich text example',
+      'field_oe_text_long' => [
+        'value' => $text_original,
+        'format' => 'filtered_html',
+      ],
+    ]);
+    $paragraph->save();
+
+    // Testing: Rich text.
```
Comment (drishu) [inline tests/src/Kernel/Paragraphs/RichTextTest.php]: this comment is not needed

File: tests/src/Kernel/Paragraphs/ParagraphsTestBase.php [hunk 9]
```diff
@@ -18,6 +18,7 @@ abstract class ParagraphsTestBase extends AbstractKernelTestBase {
   protected static $modules = [
     'allowed_formats',
     'datetime',
+    'editor',
```
Comment (drishu) [inline tests/src/Kernel/Paragraphs/ParagraphsTestBase.php]: this module is not needed

**Review by drishu — CHANGES_REQUESTED** [review]

**Review by drishu — APPROVED** [review]

**Review by escuriola — APPROVED** [review]

**Pull request merged**